### PR TITLE
Add rotary encoder support

### PR DESCRIPTION
**Pull request by mmoskal**

### CODE REVIEW

**Comment by pelikhan** [comment]
I will add the sim stuff next week

**Comment by mmoskal** [comment]
well, it needs blocks as well

File: libs/base/configkeys.h [hunk 1]
```diff
@@ -95,6 +95,9 @@
 #define CFG_PIN_PWREN 77
 #define CFG_DISPLAY_TYPE 78
 
+#define CFG_ROTARY_ENCODER_PIN_A 79
+#define CFG_ROTARY_ENCODER_PIN_A 80
```
Comment (mmoskal) [inline libs/base/configkeys.h]: B?

File: libs/base/configkeys.h [hunk 2]
```diff
@@ -95,6 +95,9 @@
 #define CFG_PIN_PWREN 77
 #define CFG_DISPLAY_TYPE 78
 
+#define CFG_ROTARY_ENCODER_PIN_A 79
+#define CFG_ROTARY_ENCODER_PIN_B 80
```
Comment (mmoskal) [inline libs/base/configkeys.h]: actually I think this should be `CFG_PIN_ROTARY_ENCODER_A` so we can use `PIN(ROTARY_ENCODER_A)`

File: libs/rotary-encoder/rotary.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+/**
+ * Rotary and other encoders
+ */
+namespace encoders {
+/**
+ * Create a new rotary encoder connected to given pins
+ */
+//% blockId=inputCreateRotaryEncoder block="create rotary encoder with pins $pinA and B $pinB"
+//% weight=99
```
Comment (mmoskal) [inline libs/rotary-encoder/rotary.cpp]: can you use this ctor function in blocks if you have `fixedInstances` on it?

Comment (pelikhan) [reply libs/rotary-encoder/rotary.cpp]: no you can't so i'm not exposing it in blocks.

**Pull request merged**